src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -407,8 +407,11 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
         // destination.Run(source)
         // destination.UseEndpoints()
 
-        // Set the route builder so that UseRouting will use the WebApplication as the IEndpointRouteBuilder for route matching
-        app.Properties.Add(WebApplication.GlobalEndpointRouteBuilderKey, _builtApplication);
+        // Set the route builder so that UseRouting will use the RouteGroupBuilder managed
+        // by the WebApplication as the IEndpointRouteBuilder instance
+        var globalRouteGroupBuilder = _builtApplication.Properties[WebApplication.GlobalEndpointRouteBuilderKey];
+        app.Properties.Add(WebApplication.GlobalEndpointRouteBuilderKey, globalRouteGroupBuilder);
+        app.Properties.Add(WebApplication.GlobalRouteGroupBuilderKey, _builtApplication.Properties[WebApplication.GlobalRouteGroupBuilderKey]);
```

Comment:
extreme nit: two different coding patterns used here, local var for property value to assign to dictionary vs. inline property value.